test/unit/locationCache.spec.ts

```diff
@@ -127,7 +128,7 @@ describe("Location Cache", function() {
           operationType: OperationType.Read,
           resourceType: ResourceType.item,
           retryCount: 0
-        });
+        } as RequestContext);
```

Comment:
I'm not sure the best way around this. LocationCache.resolveServiceEndpoint expects the full RequestContext type which is why I'm asserting. I could change it to accept a partial type just for this test? 

```
resolveServiceEndpoint(request: Pick<RequestContext, 'operationType' | 'resourceType' | 'retryCount' >)
```

Feels a little weird to do this just to avoid type asserting in the test. But it does make all the types happy.

Reply:
I like the partial because it's a lot more explicit about what it expects. I don't like it because it's ugly, but it's internal. I think I lean towards the partial type.

Reply:
Pick<> it is